shared-libs/rules-engine/src/provider-wireup.js

```diff
@@ -173,8 +173,11 @@ module.exports = {
     }
 
     // this function accepts subject ids, but rulesStateStore accepts a contact id, so a conversion is required
-    return provider.contactsBySubjectId(subjectIds)
-      .then(contactIds => rulesStateStore.markDirty(contactIds));
+    return enqueue(() => {
+      return provider
+        .contactsBySubjectId(subjectIds)
+        .then(contactIds => rulesStateStore.markDirty(contactIds));
+    });
```

Comment:
This fixes a very common feedback doc:
`Possibly unhandled rejection: {"status":409,"name":"conflict","message":"Document update conflict","error":true}`

 https://github.com/medic/cht-core/issues/7148